mitodl/mit-open
# dev mode for search

### What are the relevant tickets?
closes https://github.com/mitodl/hq/issues/4984
closes https://github.com/mitodl/hq/issues/5023

### Description (What does it do?)
This pr adds a dev_mode parameter to both the learning resource and content file searches. If dev_mode=True the api returns raw open search results with explain=True set. This will allow us to more easily troubleshoot search issues and understand the search results better.

It also fixes a bug that breaks searching content files by resource_id

### How can this be tested?
Search for http://localhost:8063/api/v1/content_file_search/?dev_mode=True&q=nutrition
Search for http://localhost:8063/api/v1/learning_resources_search/?dev_mode=True&q=nutrition

Verify that both searches work and you see the raw open search results and explanation.

Additionally try searching for
http://localhost:8063/api/v1/content_file_search/?resource_id=3669
this should not throw an error

Finally,
http://localhost:8062/search should work the same as previously

### Additional Context
One of the motivating reasons for this PR was to see why "Nuclear Energy: Science, Systems and Society" was showing up when you search for "nutrition". This turned out to be an ok result - that course had a two content files that addressed the consequences of nuclear poisoning on the gi track. The explain function does not actually give a good explanation when the score comes from content files, all you get is
```
 {
            "value": 8.823367,
             "description": "A match, join value 3669",
              "details": []
 },
```
to let you know that the score is coming from the join but
Search for http://localhost:8063/api/v1/learning_resources_search/?dev_mode=True&q=nutrition&resource_id=3669 returns more details about where the match is coming from
